### ISSUE TITLE
Multi-scale cropping based on density maps.

### ISSUE THREAD

**Issue by bw4sz** [weecology/DeepForest#619]
https://openaccess.thecvf.com/content_CVPRW_2020/papers/w11/Li_Density_Map_Guided_Object_Detection_in_Aerial_Images_CVPRW_2020_paper.pdf

<img width="836" alt="image" src="https://github.com/weecology/DeepForest/assets/1208492/83af5844-7748-485b-91e0-791e1ecc9e66">

We often have multi-scale challenges, especially as predict_tile just cuts images into uniform pieces. This an interesting idea for bird detection. with trees the images are often uniformly trees.

**Comment by bw4sz** [comment]
Sparse RCNN 

https://github.com/PeizeSun/SparseR-CNN
worth exploring as well.

**Comment by Om-Doiphode** [comment]
Hi @bw4sz, can I work on this issue?

**Comment by henrykironde** [comment]
@Om-Doiphode, Feel free to address any unresolved issues without needing explicit permission. If an issue remains unassigned, you are welcome to take initiative and work on it.

**Comment by Om-Doiphode** [comment]
I have read the research paper on `Density Map Guided Object Detection in Aerial Images`. Shoudl I include the code for `DMNet` under `deepforest/models` folder?

**Comment by ethanwhite** [comment]
This seems like a pretty complex topic that is unlikely to be easily added on a short time-frame, but could make a decent GSOC proposal. This would implement dynamic cropping, which would influence the behavior of `predict_tile` rather than replacing the detection model (though I'm not super clear on the double use of the detector yet).

@bw4sz - I'm assuming you were thinking about this as a more complex feature request project rather than a smaller issue. Is that right?

**Comment by bw4sz** [comment]
yes, anything with the machine learning label is going to be a pretty big lift.

**Comment by ethanwhite** [comment]
Thanks @bw4sz.

So, @Om-Doiphode - it's great that you're excited about this one, but we think it's better to think of it as a GSOC project to propose (if you want; there are certainly other good options) not an initial contribution. We've done some issue cleanup and labeled [a bunch of additional issues as good first issue](https://github.com/weecology/DeepForest/labels/good%20first%20issue) which will be a little more manageable to engage with as you're getting familiar with the code base.

**Comment by Om-Doiphode** [comment]
Thank you @ethanwhite
